MaxText/layers/linears.py

```diff
@@ -376,22 +367,43 @@ def gmm(inputs, kernel, group_sizes):
 
       inputs = inputs.astype(self.dtype)
       kernel = kernel.astype(self.weight_dtype)
-
-      output = mblx.gmm(lhs=inputs,
-                        rhs=kernel,
+      output = mblx.gmm(lhs=inputs, 
+                        rhs=kernel, 
                         group_sizes=group_sizes,
-                        tiling=(512, 512, 512))
-
+                        preferred_element_type=jnp.bfloat16,
+                        tiling=tile_size)
+      
       if hs_shape[0] % pad_length:
         output = output[:hs_shape[0]]
       return output
 
-    layer_w0 = gmm(inputs, w0_kernel, group_sizes)
-    layer_w1 = gmm(inputs, w1_kernel, group_sizes)
-    layer_act = _convert_to_activation_function(mlp_activation)(layer_w0)
-    intermediate_layer = jnp.multiply(layer_act, layer_w1)
-    output = gmm(intermediate_layer, wo_kernel, group_sizes)
-    return output
+    # Currently, we only support data parallelism with Megablox (sharding on batch dimensions)
+    @functools.partial(
+        shard_map.shard_map,
+        mesh=self.mesh,
+        in_specs=(
+              (nn.logical_to_mesh_axes(("activation_batch", None, None))),
```

Comment:
Nit: prefer using the common types https://github.com/google/maxtext/blob/fe4bfdcbff98877dfd611c8f802ebf94d6a9a06e/MaxText/common_types.py#L36 alias to avoid specifying strings (which could have silent typo errors)

Reply:
sg, I will update that.